hydra-node/src/Hydra/Chain/Blockfrost/Client.hs

```diff
@@ -63,153 +83,83 @@ runBlockfrostM prj action = do
   SigningKey PaymentKey ->
-  IO [TxId]
-publishHydraScripts projectPath sk = do
-  prj <- Blockfrost.projectFromFile projectPath
-  runBlockfrostM prj $ do
-    Blockfrost.Genesis
-      { _genesisNetworkMagic = networkMagic
-      , _genesisSystemStart = systemStart'
-      } <-
-      queryGenesis
-    pparams <- toCardanoPParams
-    let address = Blockfrost.Address (vkAddress networkMagic)
-    let networkId = toCardanoNetworkId networkMagic
-    let changeAddress = mkVkAddress networkId vk
-    stakePools' <- Blockfrost.listPools
-    let stakePools = Set.fromList (toCardanoPoolId <$> stakePools')
-    let systemStart = SystemStart $ posixSecondsToUTCTime systemStart'
-    eraHistory <- mkEraHistory
-    utxo <- Blockfrost.getAddressUtxos address
-    let cardanoUTxO = toCardanoUTxO utxo changeAddress
-
-    txs <- liftIO $ buildScriptPublishingTxs pparams systemStart networkId eraHistory stakePools cardanoUTxO sk
-    forM txs $ \(tx :: Tx) -> do
-      void $ Blockfrost.submitTx $ Blockfrost.CBORString $ fromStrict $ serialiseToCBOR tx
-      pure $ txId tx
- where
-  vk = getVerificationKey sk
-
-  vkAddress networkMagic = textAddrOf (toCardanoNetworkId networkMagic) vk
-
-scriptTypeToPlutusVersion :: Blockfrost.ScriptType -> Maybe Language
-scriptTypeToPlutusVersion = \case
-  Blockfrost.PlutusV1 -> Just PlutusV1
-  Blockfrost.PlutusV2 -> Just PlutusV2
-  Blockfrost.PlutusV3 -> Just PlutusV3
-  Blockfrost.Timelock -> Nothing
-
-toCardanoPoolId :: Blockfrost.PoolId -> Hash StakePoolKey
-toCardanoPoolId (Blockfrost.PoolId textPoolId) =
-  case deserialiseFromRawBytesHex (encodeUtf8 textPoolId) of
-    Left err -> error (show err)
-    Right pool -> pool
-
-toCardanoUTxO :: [Blockfrost.AddressUtxo] -> AddressInEra -> UTxO' (TxOut CtxUTxO)
-toCardanoUTxO utxos addr = UTxO.fromList (toEntry <$> utxos)
- where
-  toEntry :: Blockfrost.AddressUtxo -> (TxIn, TxOut CtxUTxO)
-  toEntry utxo = (toCardanoTxIn utxo, toCardanoTxOut utxo addr)
-
-toCardanoTxIn :: Blockfrost.AddressUtxo -> TxIn
-toCardanoTxIn Blockfrost.AddressUtxo{_addressUtxoTxHash = Blockfrost.TxHash{unTxHash}, _addressUtxoOutputIndex} =
-  case deserialiseFromRawBytesHex (encodeUtf8 unTxHash) of
-    Left err -> error (show err)
-    Right txid -> TxIn txid (TxIx (fromIntegral _addressUtxoOutputIndex))
-
--- REVIEW! TxOutDatumNone and ReferenceScriptNone
-toCardanoTxOut :: Blockfrost.AddressUtxo -> AddressInEra -> TxOut CtxUTxO
-toCardanoTxOut addrUTxO addr =
-  TxOut addr (toCardanoValue _addressUtxoAmount) TxOutDatumNone ReferenceScriptNone
- where
-  Blockfrost.AddressUtxo{_addressUtxoAmount, _addressUtxoDataHash, _addressUtxoInlineDatum, _addressUtxoReferenceScriptHash} = addrUTxO
-
-toCardanoPolicyId :: Text -> PolicyId
-toCardanoPolicyId pid =
-  case deserialiseFromRawBytesHex (encodeUtf8 pid) of
-    Left err -> error (show err)
-    Right p -> p
+  BlockfrostClientT IO [TxId]
+publishHydraScripts sk = do
```

Comment:
Should: As we have a nice interface to the specifics of accessing the cardano chain (currently a type class), we should be able to write a function `publishHydraScripts :: ChainBackend b => b -> SigningKey PaymentKey -> IO [TxId]`